### PR TITLE
Add backbone

### PR DESCRIPTION
**Pull request by Wokarol**
This will act as the backbone for managing global references and data

### CODE REVIEW

File: SpaceScrapper/Assets/_Game/_Scripts/Runtime/Global/Game.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace SpaceScrapper
+{
+    /// <summary>
+    /// Backbone of the Game
+    /// </summary>
+    public static class Game
+    {
+        public static SceneContext SceneContext { get; private set; }
+        public static GameEvents Events { get; private set; }
```
Comment (Toreole) [inline SpaceScrapper/Assets/_Game/_Scripts/Runtime/Global/Game.cs:13]: Why even go through Game.GameEvents, when we could just go directly to GameEvents tho

Comment (Wokarol) [reply SpaceScrapper/Assets/_Game/_Scripts/Runtime/Global/Game.cs:13]: Centralises where we keep the reference and allows us to safely reset it in the method which will be handy for later when we will disable domain reload on playmode enter

**Review by zenvin-dev — APPROVED** [review]

**Review by Toreole — APPROVED** [review]

File: SpaceScrapper/Assets/_Game/_Scripts/Runtime/Global/SceneContext.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+                Value = null;
+            }
+
+            public static implicit operator T(Handle<T> or)
+            {
+                return or.Value;
+            }
+
+            private string GetName(T o)
```
Comment (oliverbooth) [inline SpaceScrapper/Assets/_Game/_Scripts/Runtime/Global/SceneContext.cs:46]: This method (`GetName`) could be made static, it's not accessing any instance members here

**Review by zenvin-dev — APPROVED** [review]

**Review by Toreole — APPROVED** [review]
i thought i literally already reviewed this yesterday

File: SpaceScrapper/Assets/_Game/_Scripts/Runtime/Global/SceneContext.cs (hunk 2 above)
Comment (oliverbooth) [reply SpaceScrapper/Assets/_Game/_Scripts/Runtime/Global/SceneContext.cs:46]: (I forgot to submit the review. Ignore my dumbass)

**Review by nt314p — APPROVED** [review]

**Review by zenvin-dev — APPROVED** [review]

**Pull request merged**